### PR TITLE
add services from other env and region to rules

### PR DESCRIPTION
**Pull request by kinarashah**
[rancher/rancher#9839](https://github.com/rancher/rancher/issues/9839)

### CODE REVIEW

File: controller/rancher/rancher.go [hunk 1]
```diff
@@ -492,6 +526,14 @@ func (lbc *LoadBalancerController) processSelector(lbMeta *LBMetadata) error {
 			continue
 		}
 
+		regionName := lbRule.Region
+		envName := lbRule.Environment
+
+		svcs, err := lbc.GetRegionServices(&svcs, regionName, envName)
+		if err != nil {
+			logrus.Warnf("couldn't find services from region %v", err)
```
Comment (alena1108) [inline controller/rancher/rancher.go]: should we return on error?

File: controller/rancher/rancher.go [hunk 2]
```diff
@@ -464,6 +477,27 @@ func (lbc *LoadBalancerController) GetLBConfigs() ([]*config.LoadBalancerConfig,
 	return lbc.BuildConfigFromMetadata(lbSvc.Name, lbSvc.EnvironmentUUID, selfHostUUID, localServicePreference, lbMeta)
 }
 
+func (lbc *LoadBalancerController) GetRegionServices(svcs *[]metadata.Service, regionName string, envName string) ([]metadata.Service, error) {
```
Comment (alena1108) [inline controller/rancher/rancher.go]: can we name svcs "localServices" for clarity?

File: controller/rancher/rancher.go (hunk 1 above)
Comment (kinarashah) [reply controller/rancher/rancher.go]: There could be other PortRules that we might need to consider, so can't return?  (will add a continue instead though)

Comment (alena1108) [reply controller/rancher/rancher.go]: @kinarashah continue

**Comment by alena1108** [comment]
LGTM. Would have to change go-rancher-metadata bindings before merging.

**Pull request merged**